core/storage/face/generic_iterator.hpp

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_GENERIC_ITERATOR_HPP
+#define KAGOME_GENERIC_ITERATOR_HPP
+
+#include <memory>
+
+namespace kagome::face {
+
+  /**
+   * An interface for a forward iterator
+   * @tparam Container over which the iterator would iterate
+   */
+  template <typename Container>
+  class GenericIterator {
```

Comment:
if it is a forward iterator, then rename to be so: `GenericForwardIterator`

Reply:
I think I will leave ForwardIterator for a class in generic_list.hpp and change the comment here. It is true that it is more dependant on the implementation if it's forward or not.